src/rascal/structure_managers/property_typed.hh

```diff
@@ -319,7 +319,11 @@ namespace rascal {
      * Accessor for last pushed entry for dynamically sized properties
      */
     reference back() {
-      auto && index{this->values.size() - this->get_nb_comp()};
+      // -1 is correction for 0-start indexing
+      auto && index{this->values.size() / this->get_nb_comp() - 1};
```

Comment:
Previous code was `this->values.size() - this->get_nb_comp()`, now it is `this->values.size() / this->get_nb_comp() - 1`. Was the previous version completely bugged? The new one is very different.

Reply:
My rationale was the following.
The previous one tried to get the index from the back, which did not correct for the zero offset. 
What I now check is how many entries are in the vector and then go back one entry. The access of the values is then `index * nb_comp`.

Reply:
Oh, OK. The indexing code below did not changed and was already using `index * nb_comp`, so I guess it was very buggy.

Reply:
This is slightly worrying me, since the behaviour changed a lot here but tests still pass. I guess this part was not tested.